### DISCUSSION TITLE
Custom Provider Support

### DISCUSSION THREAD

**Discussion by hbjydev** [keycloak/keycloak#12981] [Archive]
Is there a way I can deploy a custom Identity Provider jar into Keycloak as deployed by the new operator? I'm trying to integrate with discord but they don't implement OpenID so I need to use a provider for it. Keycloak always adds openid as a scope but that breaks the login attempt if I try it manually

There used to be an 'extensions' key in the KeycloakSpec type, back on the old operator, which would download and deploy it based on a URL. Could we do something similar here?

**Comment by vmuzikar** [comment]
@hbjydev Hello, for the new Operator we encourage the users to build a custom Keycloak image should they want to extend it. This approach follows cloud-native best practices for immutability and on top of that allows you to build a pre-augmented/optimized image with super-fast startup times. Please see also: https://www.keycloak.org/operator/customizing-keycloak
